.github/workflows/style_check.yaml

```diff
@@ -0,0 +1,39 @@
+name: Style Checks
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+
+jobs:
+  style_check:
+    name: Style Checks
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout current git repository
+      - name: Checkout
+        uses: actions/checkout@v2.3.1
+
+      # Setup Miniconda
+      - name: Setup Miniconda
+        uses: goanpeca/setup-miniconda@v1.6.0
+        with:
+          python-version: 3.7
+          channels: conda-forge
```

Comment:
Should we just use pip to install black, flake8 and pylint?

Reply:
Yes that would be much faster.

Reply:
I was thinking of using [Github's Super Linter](https://github.blog/2020-06-18-introducing-github-super-linter-one-linter-to-rule-them-all/) (that lints all files, including non-python ones), but they don't seem to support `flake8` (see https://github.com/github/super-linter/issues/123) or `black` yet (see https://github.com/github/super-linter/issues/226). So will stick to manually installing it from `pip` for now.

Reply:
> Yes that would be much faster.

Wow, the Style Checks now run in 23s (using pip install), instead of 57s (with conda install). For reference, Azure Pipelines took 39s (though we could have used pip instead of conda before). I had to double check that it actually ran properly!